src/main/java/com/google/devtools/build/lib/analysis/config/RunUnder.java

```diff
@@ -13,36 +13,33 @@
 // limitations under the License.
 package com.google.devtools.build.lib.analysis.config;
 
+import com.google.common.collect.ImmutableList;
 import com.google.devtools.build.lib.cmdline.Label;
-import java.util.List;
 
 /** Components of the {@code --run_under} option. */
-public interface RunUnder {
+public sealed interface RunUnder {
   /**
    * @return the whole value passed to --run_under option.
    */
-  String getValue();
+  String value();
 
   /**
-   * Returns label corresponding to the first word (according to shell
-   * tokenization) passed to --run_under.
-   *
-   * @return if the first word (according to shell tokenization) passed to
-   *         --run_under starts with {@code "//"} returns the label
-   *         corresponding to that word otherwise {@code null}
+   * @return everything except the first word (according to shell tokenization) passed to
+   *     --run_under.
    */
-  Label getLabel();
+  ImmutableList<String> options();
 
   /**
-   * @return if the first word (according to shell tokenization) passed to
-   *         --run_under starts with {@code "//"} returns {@code null}
-   *         otherwise the first word
+   * @param label if the first word (according to shell tokenization) passed to --run_under starts
+   *     with {@code "//"} returns the label corresponding to that word otherwise {@code null}
```

Comment:
This is a bit clunky. Get rid of the `@param` and "returns" text, and just explain how a `LabelRunUnder` and `CommandRunUnder` are different.

Also I'm pretty sure the "leading `\\`" part is wrong, since this is just using a `LabelConverter`, so anything label-shaped will work.